File: 15_10_2024_Subarray_Range_With_Given_Sum.cpp
class Solution {
  public:
    // Function to count the number of subarrays which adds to the given sum.
    int subArraySum(vector<int>& arr, int tar) {
        // Your code here
        int n = arr.size();
        unordered_map<int, int> summap;
        int sum = 0;
        int count = 0;
        summap[0] = 1;
        for (int i = 0; i < n; i++) {
            sum += arr[i];
            int rem=sum - tar;
            count += summap[rem];
            summap[sum]+=1;
        }
        return count;
    }
};

File: 15_10_2024_Subarray_Range_With_Given_Sum.py
#User function Template for python3

class Solution:
    
    #Complete this fuction
    #Function to count the number of subarrays which adds to the given sum.
    def subArraySum(self,arr, tar):
        hash = {0 : 1}
        res = c_sum = 0
        for i in range(len(arr)):
            c_sum += arr[i]
            if c_sum - tar in hash:
                res += hash[c_sum - tar]
            hash[c_sum] = hash.get(c_sum, 0) + 1
        
        return res
        
        
